geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/AbstractFileSystemStorage.scala

```diff
@@ -67,11 +67,13 @@ abstract class AbstractFileSystemStorage(
   /**
     * Create a writer for the given file
     *
+    * @param partition the partition that the file belongs to
+    * @param action whether to append or modify
     * @param file file to write to
     * @param observer observer to report stats on the data written
     * @return
     */
-  protected def createWriter(file: Path, observer: FileSystemObserver): FileSystemWriter
+  protected def createWriter(partition: String, action: StorageFileAction, file: Path, observer: FileSystemObserver): FileSystemWriter
```

Comment:
```suggestion
  protected def createWriter(partition: String, action: StorageFileAction, file: Path, observer: Option[FileSystemObserver]): FileSystemWriter
```
since we don't have the updateObserver anymore, we should observer an option. that will let us clean up the code down on line 239 as well, we won't need the NoOpObserver any more.

Reply:
i think that means we can get rid of the BoundsObserver and related changes now too?